### PR TITLE
Cleanup and travis

### PR DESCRIPTION
**Pull request by grassator**
The PR is better reviewed commit-by-commit

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,11 +18,11 @@
   "author": "Team Pathfinder <team-pathfinder@zalando.de>",
   "license": "MIT",
   "devDependencies": {
-    "eslint": "2.10.2",
-    "jsdoc-to-markdown": "1.3.6",
-    "mocha": "2.4.5",
-    "proxyquire": "1.7.9",
-    "sinon": "1.17.4"
+    "eslint": "3.19.0",
```
Comment (vigneshshanmugam) [inline package.json]: shall we switch to package.lock/shrinkwrap or yarn.lock to avoid these

Comment (grassator) [inline package.json]: Created an issue for this https://github.com/zalando-incubator/perron/issues/22

File: .eslintrc.json [hunk 2]
```diff
@@ -4,36 +4,275 @@
...
-    "parserOptions": {
-        "ecmaFeatures": {
-            "experimentalObjectRestSpread": true
-        }
-    },
     "rules": {
-        "no-console": [
-            "warn"
+        "accessor-pairs": "error",
+        "array-callback-return": "error",
```
Comment (vigneshshanmugam) [inline .eslintrc.json]: why do we need to reset all these rules if  we use `eslint:recommended `

Comment (grassator) [inline .eslintrc.json]: This config is much more strict and it was generated from the code

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+language: node_js
+env: CI=true
+sudo: false
+
+node_js:
```
Comment (vigneshshanmugam) [inline .travis.yml]: node_modules is not cached by default in Travis yet, so we can specify 
```
cache:
  directories:
    - node_modules
```

Comment (grassator) [inline .travis.yml]: NPM install takes a couple of seconds on Travis. Do we really need it?

Comment (vigneshshanmugam) [inline .travis.yml]: not sure though, we have lots of repos on travis in zalando org so this might help.

Comment (grassator) [inline .travis.yml]: Added

**Pull request merged**